gtsam/hybrid/HybridGaussianFactorGraph.cpp

```diff
@@ -96,8 +109,15 @@ GaussianMixtureFactor::Sum sumFrontals(
       }
 
     } else if (f->isContinuous()) {
-      deferredFactors.push_back(
-          boost::dynamic_pointer_cast<HybridGaussianFactor>(f)->inner());
+      // Check if f is HybridConditional or HybridGaussianFactor.
+      if (auto hc = boost::dynamic_pointer_cast<HybridConditional>(f)) {
```

Comment:
This should just work without explicit casting as `HybridConditional` is a subclass of HybridGaussianConditional

Reply:
Yeah I don't remember why I did this. Guess I was debugging something.